xvec/accessor.py

```diff
@@ -918,6 +918,274 @@ def to_geodataframe(
+        n_jobs : int, optional
+            Number of parallel threads to use.
+
+
+        Returns
+        -------
+        Dataset
+            A subset of the original object with N-1 dimensions indexed by
+            the the GeometryIndex.
+
+        """
+
+        try:
+            import geopandas as gpd
+        except ImportError as err:
+            raise ImportError(
+                "The geopandas package is required for `xvec._spatial_agg()`. "
+                "You can install it using 'conda install -c conda-forge geopandas' or "
+                "'pip install geopandas'."
+            ) from err
+
+        try:
+            import rioxarray  # noqa
+        except ImportError as err:
+            raise ImportError(
+                "The rioxarray package is required for `xvec._spatial_agg()`. "
+                "You can install it using 'conda install -c conda-forge rioxarray' or "
+                "'pip install rioxarray'."
+            ) from err
+
+        try:
+            from joblib import Parallel, delayed
+        except ImportError as err:
+            raise ImportError(
+                "The joblib package is required for `xvec._spatial_agg()`. "
+                "You can install it using 'conda install -c conda-forge joblib' or "
+                "'pip install joblib'."
+            ) from err
+
+        try:
+            from tqdm import tqdm
+        except ImportError as err:
+            raise ImportError(
+                "The tqdm package is required for `xvec._spatial_agg()`. "
+                "You can install it using 'conda install -c conda-forge tqdm' or "
+                "'pip install tqdm'."
+            ) from err
+
+        import gc
+
+        transform = self._obj.rio.transform()
+        geometry_chunks = [
+            geometries[i : i + chunk_size]
+            for i in range(0, len(geometries), chunk_size)
+        ]
+
+        stats_dic = {}
+        for var in self._obj.data_vars:
+            stats_dic[var] = []
+
+            computed_results = []
+            for chunk in tqdm(geometry_chunks):
+                # Create a list of delayed objects for the current chunk
+                chunk_results = Parallel(n_jobs=n_jobs)(
+                    delayed(self._agg_geom)(
+                        geom,
+                        transform,
+                        var,
+                        x_coords,
+                        y_coords,
+                        stat=stat,
+                    )
+                    for geom in chunk
+                )
+                computed_results.extend(chunk_results)
+            stats_dic[var] = computed_results
+
+            # Clean the space
+            gc.collect()
+
+        # Unpack the results into VectorCube
+        df = pd.DataFrame()
+        keys_items = {}
+        for k in stats_dic.keys():
+            s = stats_dic[k]
+            columns = []
+            for t in range(len(self._obj.time)):
```

Comment:
Why do we have `time` hard-coded here? This should be agnostic to other dimensions. The same hard-coding is also below.

Reply:
That's correct. I did it based on the example of raster cube. I'll make the necessary adjustments to ensure it's dimension-agnostic.